state/factory/workingset_test.go

```diff
@@ -0,0 +1,161 @@
+package factory
+
+import (
+	"context"
+	"errors"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/pkg/unit"
+	"github.com/iotexproject/iotex-core/test/identityset"
+	"github.com/iotexproject/iotex-core/testutil"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
+	"time"
+	//"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	//"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+func TestWriteReadView(t *testing.T) {
+	require := require.New(t)
+	sf, err := NewFactory(config.Default, InMemTrieOption())
+	require.NoError(err)
+	ctx := protocol.WithBlockchainCtx(
+		protocol.WithRegistry(context.Background(), protocol.NewRegistry()),
+		protocol.BlockchainCtx{Genesis: config.Default.Genesis},
+	)
+	require.NoError(sf.Start(ctx))
+	ws, err := sf.(workingSetCreator).newWorkingSet(ctx, 1)
+
+	t.Run("ReadView call error", func(t *testing.T) {
+		retrdf, retErr := ws.ReadView("")
+		expectedErrors := errors.New("name : name does not exist")
+		require.EqualError(retErr, expectedErrors.Error())
+		require.Equal(retrdf, nil)
+	})
+
+	t.Run("WriteView ReadView normal call ", func(t *testing.T) {
+		require.NoError(ws.WriteView("test", "test"))
+		retrdf, retErr := ws.ReadView("test")
+		require.NoError(retErr)
+		require.Equal(retrdf, "test")
+	})
+
+	t.Run("Unload call error", func(t *testing.T) {
+		retrdf, retErr := ws.Unload("")
+		expectedErrors := errors.New("name : name does not exist")
+		require.EqualError(retErr, expectedErrors.Error())
+		require.Equal(retrdf, nil)
+	})
+
+	t.Run("Load Unload normal call ", func(t *testing.T) {
```

Comment:
think we could just write an all-in-one test for these?
Unload, ReadView will fail at first
do a WriteView/Load
then Unload, ReadView would success
Unload, ReadView of a non-existing name still fail

multiple t.Run() is OK, but feels like overkill since each test is about 6 lines